src/main/java/gift/repository/ProductRepositoryImpl.java

```diff
@@ -0,0 +1,62 @@
+package gift.repository;
+
+import gift.entity.Product;
+import org.springframework.stereotype.Repository;
+
+import java.util.HashMap;
+import java.util.List;
+
+@Repository
+public class ProductRepositoryImpl implements ProductRepository {
+
+    // 컬렉션을 활용하여 임시로 구현
+    final private HashMap<Long, Product> productMap;
```

Comment:
```suggestion
    final private Map<Long, Product> productMap;
```
HashMap의 인터페이스인 Map으로 선언해도 좋을 것 같아요 😄